### PR TITLE
Replace mapfile with kube::util::read-array

### PR DESCRIPTION
**Pull request by oomichi**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

During the review of https://github.com/kubernetes/kubernetes/pull/78779
we've known mapfile doesn't work on macOS.
So we need to use alternative way instead and this is it.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by odinuge — APPROVED** [review]
LGTM :smile:

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -349,23 +349,23 @@ cleanup()
   # fi
 
   # Check if the API server is still running
-  [[ -n "${APISERVER_PID-}" ]] && mapfile -t APISERVER_PIDS < <(pgrep -P "${APISERVER_PID}" ; ps -o pid= -p "${APISERVER_PID}")
+  [[ -n "${APISERVER_PID-}" ]] && kube::util::read-array APISERVER_PIDS < <(pgrep -P "${APISERVER_PID}" ; ps -o pid= -p "${APISERVER_PID}")
```
Comment (BenTheElder) [inline hack/local-up-cluster.sh]: This should just work but FYI @dims 😅

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Pull request merged**